modules/tf_cloudbuild_core/main.tf

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cloudbuild_project_id = var.project_id != "" ? var.project_id : format("%s-%s", var.project_prefix, "cloudbuild")
+
+  cloudbuild_apis = [
+    "cloudbuild.googleapis.com",
+    "sourcerepo.googleapis.com",
+    "storage-api.googleapis.com",
+    "iam.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "cloudbilling.googleapis.com"
+  ]
+
+  activate_apis = distinct(concat(var.activate_apis, local.cloudbuild_apis))
+}
+
+resource "random_id" "suffix" {
+  byte_length = 2
+}
+
+module "cloudbuild_project" {
+  source  = "terraform-google-modules/project-factory/google"
+  version = "~> 13.0"
+
+  name                        = local.cloudbuild_project_id
+  random_project_id           = var.use_random_suffix
+  disable_services_on_destroy = false
+  folder_id                   = var.folder_id
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  activate_apis               = local.activate_apis
+  labels                      = var.project_labels
+}
+
+module "cloudbuild_bucket" {
+  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
+  version = "~> 3.2"
+
+  name          = "${module.cloudbuild_project.project_id}_cloudbuild"
```

Comment:
What do we use this bucket for?

Reply:
Got it, lets add a comment here with any upstream docs if any.

Reply:
added an explanation
@bharathkkb could you PTAL?